diff --git a/nuttx/drivers/lcd/st7735.c b/nuttx/drivers/lcd/st7735.c
index 7c713b4..f3aca61 100644
--- a/nuttx/drivers/lcd/st7735.c
+++ b/nuttx/drivers/lcd/st7735.c
@@ -204,5 +204,7 @@ static void st7735_bpp(FAR struct st7735_dev_s *dev, int bpp);
 static void st7735_wrram(FAR struct st7735_dev_s *dev,
                          FAR const uint16_t *buff, size_t size);
+#ifndef CONFIG_LCD_NOGETRUN
 static void st7735_rdram(FAR struct st7735_dev_s *dev,
                          FAR uint16_t *buff, size_t size);
+#endif
 static void st7735_fill(FAR struct st7735_dev_s *dev, uint16_t color);
@@ -492 +494,2 @@ static void st7735_wrram(FAR struct st7735_dev_s *dev,
+#ifndef CONFIG_LCD_NOGETRUN
 /****************************************************************************
@@ -508 +511,2 @@ static void st7735_rdram(FAR struct st7735_dev_s *dev,
 }
+#endif
@@ -557 +561 @@ static int st7735_putrun(FAR struct lcd_dev_s *dev,
-  ginfo("row: %d col: %d npixels: %d\n", row, col, npixels);
+  ginfo("row: %d col: %d npixels: %zu\n", row, col, npixels);
@@ -588 +592 @@ static int st7735_getrun(FAR struct lcd_dev_s *dev,
-  ginfo("row: %d col: %d npixels: %d\n", row, col, npixels);
+  ginfo("row: %d col: %d npixels: %zu\n", row, col, npixels);
